### PR TITLE
Text book implementation of linked lists in GoLang

### PR DESCRIPTION
**Pull request by lborg019**
No need to reinvent the wheel.
[Source](https://golang.org/src/container/list/list.go)

### CODE REVIEW

**Comment by lborg019** [comment]
I'll fix up everything according to the guidelines.

File: heapsort/heapsort.go [hunk 1]
```diff
@@ -0,0 +1,63 @@
+package main
```
Comment (Ch3ck) [inline heapsort/heapsort.go]: this should be package heapsort

File: quick-sort.go [hunk 2]
```diff
@@ -2,12 +2,12 @@ package main
```
Comment (Ch3ck) [inline quick-sort.go]: package should be named accordingly

File: heapsort/heapsort.go [hunk 3]
```diff
@@ -1,5 +1,7 @@
 package main
```
Comment (Ch3ck) [inline heapsort/heapsort.go]: Write this as a heapsort package not main

File: heapsort/heapsort.go [hunk 4]
```diff
@@ -48,10 +50,34 @@ func Heapsort(A []int, N int) {
 }
 
 func main() {
```
Comment (Ch3ck) [inline heapsort/heapsort.go]: write a package not a main function. Make sure you include a heapsort_test.go

**Pull request merged**